org.lflang/src/org/lflang/generator/c/CGenerator.java

```diff
@@ -547,6 +548,29 @@ public void doGenerate(Resource resource, LFGeneratorContext context) {
             Exceptions.sneakyThrow(e);
         }
 
+        // Create a .vscode/settings.json file in the target directory so that VSCode can
+        // immediately compile the generated code.
+        try {
+            String compileDefs = targetConfig.compileDefinitions.keySet().stream()
+                .map(key -> "\"-D" + key + "=" + targetConfig.compileDefinitions.get(key) + "\"")
+                .collect(Collectors.joining(",\n"));
+            String settings = "{\n"
+                + "\"cmake.configureArgs\": [\n"
+                + compileDefs
+                + "\n]\n}\n";
+            Path vscodePath = Path.of(fileConfig.getSrcGenPath() + File.separator + ".vscode");
+            if (!Files.exists(vscodePath))
+            Files.createDirectory(vscodePath);
+            FileUtil.writeToFile(
+                settings,
+                Path.of(fileConfig.getSrcGenPath()
+                    + File.separator + ".vscode"
+                    + File.separator + "settings.json")
+            );
+        } catch (IOException e) {
+            Exceptions.sneakyThrow(e);
```

Comment:
The `sneakyThrow` pattern was inherited from generated Xtend code, and AFAIK there is no good reason to use it. If we can use the error reporter, that would be best, but just an ordinary `RuntimeException` would also do, I guess.

Reply:
Create #1761.